LanguageFeatures/Extension-types/exhaustiveness_flow_analysis_A01_t01.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Extension type erasure is not used during flow analysis
+///
+/// @description Check that an extension type erasure is not used during flow
+/// analysis
```

Comment:
Perhaps:
```suggestion
/// analysis that isn't concerned with run-time type tests
```

Reply:
I don't really understand what we're testing here (e.g., line 17-30).

We could have a variable whose type is an extension type like `ET`, and it could be erased such that it is actually treated as if it had type `int` (which would be a bug). But we don't have that.

We could also have an object pattern like `ET(value: var __)`, but in the given example there is no promotion because `ET` isn't a subtype of `int`. So `i` still has the type `int`, and there is nothing to erase (so we can't really be testing erasure).

The last perspective that I can think of right now is that the flow analysis will (and should) use the extension type erasure when it is detecting that `case ET...` handles all values of type `int`.

However, the switch isn't on an always-exhaustive type, so it isn't required to be exhaustive.

I think we're just going to have the error that `x` isn't definitely assigned in line 27, and, as far as I can see, nothing about the code so far has anything to do with "extension type erasure isn't used for non-run-time-type test flow analysis".

Reply:
It's tricky: How do we even detect that flow analysis has a bug which is that it uses the erased type for "anything other than run-time type tests", because "anything other than" is complex in the first place.

But an obvious test would be that a variable pattern like `ET v` will give `v` the type `ET` (and not `int`). This could be tested by using something in the interface of `ET`, or by an assignment to yet another variable like `ET v2 = v;`. 

We could also have a test like `switch (o) { case ET(): o.myETMethod(); ... }` where `o` has a static type that allows promotion to `ET` (e.g., `o` could have the type `Object?`). This would show that promotion of the scrutinee is capable of promoting to an extension type (and it doesn't erase the extension type when using the scrutinee in that case).

Reply:
Yes, got it!